src/dune_sexp/escape.ml

```diff
@@ -1,26 +1,92 @@
 open! Stdune
 
+module Utf8 = struct
+  (*
+     The first byte of an utf8 character gives the size in bytes of the utf8:
+
+     0xxxxxxxx -> 1
+     110xxxxxx -> 2
+     1110xxxxx -> 3
+     11110xxxx -> 4
+  *)
+  let utf8_byte_length u =
+    match Char.code u with
+    | u when u < 128 -> 1
+    | u when u < 194 -> 0
+    | u when u < 224 -> 2
+    | u when u < 240 -> 3
+    | u when u < 245 -> 4
+    | _ -> 0
+  ;;
+
+  let unsafe_get s i = String.unsafe_get s i |> Char.code
+  let next_utf8_length s i = utf8_byte_length (String.unsafe_get s i)
+
+  let is_utf8_valid s i l =
+    assert (String.length s >= l);
+    match l with
+    | 1 -> true
+    | 2 ->
+      let b1 = unsafe_get s (i + 1) in
+      if b1 lsr 6 != 0b10 then false else true
```

Comment:
Could you replace `if x then false else true` with `not x`. Here and everywhere else

Reply:
Done in https://github.com/ocaml/dune/commit/b52cef49ae33015dc202a5f4adc671063bd15234